superset/assets/src/modules/LoaderRegistry.js

```diff
@@ -0,0 +1,16 @@
+import Registry from './Registry';
+
+export default class LoaderRegistry extends Registry {
```

Comment:
would it simplify things to get rid of this class and just have the parent `Registry` class have first class support for loaders out of the box?

Reply:
especially if the `Registry` class already supports promises

Reply:
I think simple registry could be useful for sync items like metadata. We can use this for other key-value stores we may have in the future.

Reply:
Oh I meant have `Registry` support sync and async all in one class. Like take either a `loader` or a `value`, and have the same promise-based API for getting either of them so you don't have to juggle different classes and different methods like `get` and `getAsPromise`.

Reply:
I am revisiting the idea of having both in same `Registry` again. May have to add explicit flag to keep track if it is a loader or not. You might have convinced me that one class that can do both will be better.